### DISCUSSION TITLE
Using a submodule to share source for your team

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#3353] [Show and tell]
I think it is important that anyone who wants to contribute to your codebase can easily get started by cloning the development branch and having everything they need ready. To achieve this, you will need to follow some extra steps that will include the source code of Oqtane in your project and allow you to make changes to your own version of the framework if necessary.

Full Post here:  https://www.marks.wiki/p/how-to-initialize-an-oqtane-repo

**Comment by leigh-pointer** [comment]
@markdav-is I think it is really important that no one changes the framework code base! **Rule number one!**

**Reply by markdav-is** [reply]
Indeed!  We've done it very rarely and contributed back those changes.  The trick we were having was keeping all the devs on the same version of Oqtane on their local machines.  Some would pull dev and be to far ahead, some would pull master.  We also have multiple Oqtane projects running in parallel, so the submodule was a way to pin a particular release to a repo.  As the community starts evolving, I hope to learn about others workflows.
